### DISCUSSION TITLE
[Help]: How to implement Carousel Thumbnails using the available contexts and stores from `shadcn-svelte`

### DISCUSSION THREAD

**Discussion by shyakadavis** [davidjerleke/embla-carousel#809] [Help]
### Summary

Hi;

I know this question is better asked in the `shadcn-svelte` community, but I'm trying my luck here - hope it's okay.

Basically, I'm trying to achieve something like this image.

<img width="969" alt="Screenshot 2024-03-08 at 13 19 35" src="https://github.com/davidjerleke/embla-carousel/assets/87414827/0fe4475d-af29-4cb7-9d46-f140c0cff195">

I managed to create the dots using the available context and stores, such that at the end of the day, I'm able to just reach into the context and grab these:

```ts
const { handleKeyDown, scrollTo, scroll_snaps, selected_index } = getEmblaContext('<Carousel.Dots/>');
```

But implementing thumbnails similarly is eluding me.

I prepared a repo/repro of how I managed to achieve the dots, if that helps.

P.S. If you notice a way to clean it up, please let me know. (rn, it "feels" like a hack because I have terrible understanding of the context api)

Links:
- Goal: https://www.embla-carousel.com/examples/predefined/#thumbnails
- Repo: https://github.com/shyakadavis/thumbs
- Stackblitz: https://stackblitz.com/~/github.com/shyakadavis/thumbs

Thank you for the consideration.

### If applicable, which variants of Embla Carousel are relevant to this question?

- [X] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [ ] embla-carousel-vue
- [X] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by davidjerleke** [comment]
- Vote here: #770.
